src/dune_rules/super_context.ml

```diff
@@ -402,6 +402,23 @@ let foreign_flags t ~dir ~expander ~flags ~language =
   in
   Action_builder.memoize (sprintf "%s flags" name) flags
 
+let link_flags t ~dir ~expander ~use_standard_c_and_cxx_flags ~flags =
+  let env_tree = t.env_tree in
+  let default =
+    get_node env_tree ~dir >>| Env_node.link_flags |> Action_builder.memo_build_join
+  in
+  let default =
+    if use_standard_c_and_cxx_flags
+    then 
+      let open Action_builder.O in
+      let+ default = default
+      and+ flags = Cxx_flags.get_flags ~for_:Link (context t) in
+      default @ List.concat_map flags ~f:(fun f -> [ "-cclib"; f ])
```

Comment:
I feel like this is not exactly the expected behavior: `standard_cxx_flags` should be the `:standard` set of flag when expanding the `link_flags` field from the `env`. The result of this expansion is then the `:standard` set of flags for the executable stanza `link_flags` field.

This way it is possible to override the defautl cxx flags from the `link_flags` field in the `env` stanza.

```
default_cxx_flags
  |
  └-- [`:standard` for base `env` `link_flags` field expansion]
      |
      └-- [`:standard` for nested `env` `link_flags` field expansion]
          |
          :
          |
          └-- [`:standard` for `executable` `link_flags` field expansion]
```

Reply:
The logic you added depend on properties of executables, env doesn't have access to it

Reply:
Yes but can't we work around that by delaying the expansion or using a placeholder ?

I think it is important to preserve the philosophy of the [ordered set language](https://dune.readthedocs.io/en/stable/concepts.html#ordered-set-language) that allows full control over the built-in flags with the `:standard` set.

Reply:
I'd like other dune devs to take a look at this.
Delaying evaluation seems to be a lot of work,  I don't know the codebase enough to understand the implication

I'm not sure what a placeholder would look like in practice.

@rgrinberg , @jeremiedimino  can you look at this ?

Reply:
@voodoos has been thinking about the handling of flags a lot though some of his recent work, so I trust his judgement here.

Everything is versioned in Dune, so changing things after the fact is more difficult. For this reason we should aim to do the right thing from the start, even if it is more work.

Reply:
@jeremiedimino, yet I don't how to implement such behavior. My hunch is that the placeholder doesn't work in practice and I'm not sure what are the implications of delaying the expansion of `: standard` in flags.

Reply:
I've tried something, that is duplicate link_flags and use the existing inheritance. @voodoos how does that look to you ?

Reply:
gentle ping

Reply:
It looks like a reasonable solution. There is just one thing we should check: getting the default cxx flags requires that Dune run the C-compiler and preprocess a header file to create the `_build/default/.dune/ccomp/ccomp` file. It should be good but I think we should check (via a test) that this file is not created if no cxx stubs are built.

Also we should have a test reflecting the cascading behaviour. It is a bit painful to do because the C++ flags will differ depending on the context; what I did in my tests was grepping with a `or` : 

```
  $ GCC_LF_LIB=" -cclib -lstdc++ -cclib -shared-libgcc"
  $ Clang_LF_LIB=" -cclib -lc++"
  $ Msvc_LF_LIB=""

  $ dune rules main.exe  | tr -s '\n' ' ' |
  > grep -ce "Main.cmx$GCC_LF_LIB)\|Main.cmx$Clang_LF_LIB)\|Main.cmx$Msvc_LF_LIB)"
  1
```

It is a bit fragile but works reasonably good.

Reply:
Also it might be useful for future readers to document the reason of the split in `link_flags.ml`.

Reply:
I've added more test, and fix an existing test at the same time.